app/controllers/api/v1/subnets_controller.rb

```diff
@@ -25,6 +25,7 @@ def show
       api :POST, '/subnets', 'Create a subnet'
       param :subnet, Hash, :required => true do
         param :name, String, :desc => 'Subnet name', :required => true
+        param :type, String, :desc => 'Type or protocol, Subnet::Ipv4 or Subnet::Ipv6, defaults to Subnet::Ipv4'
```

Comment:
This is quite user-unfriendly, we should avoid exposing the class names in the API if at all possible - we do it better in some places than others currently.  Take a look at ComputeResource, which has a .new_provider method called from the API controller's #create method, allowing the user to specify just "Libvirt" or similar without needing to know it's Foreman::Model::Libvirt.